ruma-events-macros/src/event_enum.rs

```diff
@@ -278,17 +287,17 @@ fn expand_any_enum_with_deserialize(
 
 fn expand_custom_variant(ident: &Ident, event_struct: &Ident) -> (TokenStream, TokenStream) {
     if ident.to_string().contains("Redacted") {
-        let redacted_event = Ident::new(&format!("Empty{}", event_struct), ident.span());
         (
             quote! {
                 /// A redacted event not defined by the Matrix specification
-                Custom(::ruma_events::#redacted_event),
+                Custom(::ruma_events::#event_struct<::ruma_events::custom::RedactedCustomEventContent>),
             },
             quote! {
                 event => {
-                    let event =
-                        ::serde_json::from_str::<::ruma_events::#redacted_event>(json.get())
-                            .map_err(D::Error::custom)?;
+                    let event = ::serde_json::from_str::<
+                        ::ruma_events::#event_struct<::ruma_events::custom::RedactedCustomEventContent>,
+                    >(json.get())
+                    .map_err(D::Error::custom)?;
```

Comment:
Is this how `rustfmt` would format this?

Reply:
Yup, that one I commented out the `quote!` and ran it.

Reply:
Okay. Looks kind of weird, but it's fine then.

Reply:
I agree I dislike the hanging `>` :shrug: